.github/workflows/image-minimizer.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * Script for minimizing big images (jpg,gif,png) when they are uploaded to GitHub and not edited otherwise
+ */
+module.exports = async ({github, context}) => {
+    const IGNORE_KEY = '<!-- IGNORE IMAGE MINIFY -->';
+    const IGNORE_ALT_NAME_END = 'ignoreImageMinify';
+    const IMG_MAX_HEIGHT_PX = 600;
+
+    // Get the body of the image
+    let initialBody = null;
+    if (context.eventName == "issue_comment") {
+        initialBody = context.payload.comment.body;
+    } else if (context.eventName == "issues") {
+        initialBody = context.payload.issue.body;
+    } else {
+        console.log("Aborting: No body found");
+        return;
+    }
+    console.log("Found body: '" + initialBody + "'");
+
+    // Check if we should ignore the currently processing element
+    if (initialBody.includes(IGNORE_KEY)) {
+        console.log('Ignoring: Body contains IGNORE_KEY');
+        return;
+    }
+
+    // Regex for finding images (simple variant) ![ALT_TEXT](https://*.githubusercontent.com/<number>/<variousHexStringsAnd->.<fileExtension>)
+    const REGEX_IMAGE_LOOKUP = /\!\[(.{0,})\]\((https:\/\/[-a-z0-9]{1,}\.githubusercontent\.com\/\d+\/[-0-9a-f]{32,}\.(jpg|gif|png))\)/gm;
+
+    // Check if we found something
+    let foundSimpleImages = REGEX_IMAGE_LOOKUP.test(initialBody);
+    if (!foundSimpleImages) {
+        console.log('Found no simple images to process');
+        return;
+    }
+
+    console.log('Found at least one simple image to process');
+
+    // Require the probe lib for getting the image dimensions
+    const probe = require('probe-image-size');
+
+    // Try to find and replace the images with minimized ones
+    let newBody = await replaceAsync(initialBody, REGEX_IMAGE_LOOKUP, async (match, g1, g2) => {
+        console.log("Found match '" + match + "'");
+        
+        if (g1.endsWith(IGNORE_ALT_NAME_END)) {
+            console.log("Ignoring match '" + match + "': IGNORE_ALT_NAME_END");
+            return match;
+        }
+        
+        let shouldModifiy = false;
+        try {
+            console.log("Probing " + g2);
+            let probeResult = await probe(g2);
+            if (probeResult == null) {
+                throw "No probeResult";
+            }
+            if (probeResult.hUnits != 'px') {
+                throw "Unexpected probeResult.hUnits (expected px but got " + probeResult.hUnits + ")";
```

Comment:
I think throwing strings directly isn't good practice in JavaScript:

```suggestion
                throw Error("Unexpected probeResult.hUnits (expected px but got " + probeResult.hUnits + ")");
```

Reply:
Thank for the hint about interpolation.
I see no advantage for the ``throw Error`` block so I chose to keep it.

Reply:
Sometimes I don't really understand JavaScript. You can do anything (syntactically) but when you run [a linter]( https://eslint.org/docs/rules/no-throw-literal) half of it is bad.

Anyway in my opinion the above code doesn't need to throw stacktraces (for now) because the exception is unique is immediately and printed to the console.

Reply:
> Sometimes I don't really understand JavaScript. You can do anything (syntactically) but when you run a linter half of it is bad.

Haha, I know, right?

> Anyway in my opinion the above code doesn't need to throw stacktraces (for now) because the exception is unique is immediately and printed to the console.

It's decided then.